### PR TITLE
Update lodash

### LINKED ISSUE #1532: Update or drop lodash due to Vulnerability
The latest version of npm has  a new audit feature to expose potential vulnerabilities. There is currently a low severity vulnerability in lodash. 

The issue is patched in lodash **v4.17.5+**

Async is currently on lodash **v4.14.0** (https://github.com/caolan/async/blob/master/package.json#L22)

More details on https://nodesecurity.io/advisories/577

I see there are also plans to drop lodash. That would fix this as well.

### PR DESCRIPTION
**Pull request by chrisdukakis**
Fixes #1532 . Updates lodash in all deps, in light of https://nodesecurity.io/advisories/577, just in case.
Pls review and merge this or #1528 , we all need it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "utility"
   ],
   "dependencies": {
-    "lodash": "^4.14.0",
+    "lodash": "^4.17.5",
     "lodash-es": "^4.14.0"
```
Comment (hargasinski) [inline package.json]: Does `lodash-es` need to be updated too?

File: package.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
     "karma-mocha-reporter": "^2.2.0",
     "mocha": "^3.1.2",
     "native-promise-only": "^0.8.0-a",
-    "nyc": "^7.0.0",
+    "nyc": "^11.8.0",
```
Comment (hargasinski) [inline package.json]: I'm assuming this is because the old version of `nyc` was using an older version of `lodash`?

**Comment by hargasinski** [comment]
Thanks for the PR! I just added a couple of quick comments.

I'm okay with merging this for a quick patch release until #1528 gets merged and released in `v3`.

File: package.json (hunk 2 above)
Comment (chrisdukakis) [reply package.json]: that's correct

File: package.json (hunk 1 above)
Comment (chrisdukakis) [reply package.json]: `lodash-es` doesn't depend on `lodash`, it's fine as is

**Comment by chrisdukakis** [comment]
Thank you @hargasinski ! I only updated packages using lodash, there are more cases that you might want to look into, but all of those are dev deps. I wasn't familiar with those tools, so I didn't push further changes and brake anything. You can try an `npm audit` and check the log.

File: package.json (hunk 1 above)
Comment (hargasinski) [reply package.json]: As far as I know, `lodash-es` is built from `lodash`. 

Either way, it's probably a good idea to keep them synced. Could you update it to `v4.17.5`? The PR will be good to go afterwards.

Comment (chrisdukakis) [reply package.json]: You are right! Done!

**Comment by aearly** [comment]
Going to merge this for a patch, will remove Lodash in 3.0

**Pull request merged**